### DISCUSSION TITLE
Overlays how-to

### DISCUSSION THREAD

**Discussion by KlimRider** [Joshua-Riek/ubuntu-rockchip#714] [Q&A]
Hi everyone.
I've read about overlays a lot, but no one describe how to get them.
For now I installed beta desktop on my opi5+ and seems that work. Not flawless but my wifi work.
These board have an annoying fan start spinning at 55 degrees.
I want to change this.
In the official how-to mention about overlays.
But I have no that files in my system (as official and with Joshua builds)
And I have no /boot/firmware/ directory.
I've added I file ubuntuEnv.txt and point to disable leds for test.
The system is ignoring this overlays settings.
I want to ask anyone's - please describe what I must Todo to get this overlays?
Everyone write that settings is in DTS files, but I have only DTD and DTBO.
As I understand this is not enough to run overlays

**Comment by KlimRider** [comment]
I found how to do this.
In the /usr/lib/firmware/<kernel version>/device-tree/rockchip/
inside this directory have files of device-tree overlays
i choose my board - rk3588-orangepi-5-plus.dtd
move this file to my home directory and convert them to the DTS by command:
dtc -I dtd -O dts -f  rk3588-orangepi-5-plus.dtd -o rk3588-orangepi-5-plus.dts
After that edit dts file as i want (change pwm-fan parameters) and convert this file back to dtd format:
dtc -I dts -O dtd -f  rk3588-orangepi-5-plus.dts -o rk3588-orangepi-5-plus.dtd
After convert move this file back to /usr/lib/firmware/<kernel version>/device-tree/rockchip/ directory and reboot the board

P.S. On my board with fan from OrangePi (special fan for this board) the PWM not work.
Guess some bug or fan cannot PWMing itself.

**Comment by KlimRider** [comment]
Maybe this will helpfull for newbies

**Comment by MattiaGrigoli** [comment]
the folder you mention (`/usr/lib/firmware//device-tree/rockchip/`) doesn't exist.
i'm using ubuntu 22.04 on orange pi 5 plus.
where could the overlays be?

**Reply by MattiaGrigoli** [reply]
found it, there was another foulder with the name of the version used.
